File: src/level4/logger.js
// Level 4: Logger (deepest)
function level4Logger() {
  return 'L4:logger';
}

function level4LogError() {
  // Flaky - sometimes fails
  if (Math.random() < 0.2) {
    throw new Error('Logger failed');
  }
  return 'L4:logError';
}

module.exports = {
  level4Logger,
  level4LogError
};

File: src/py_level4/logger.py
"""Level 4: Logger (deepest)"""
import random

def level4_logger():
    return "L4:logger"

def level4_log_error():
    """Flaky - fails 20% of the time"""
    if random.random() < 0.2:
        raise Exception("Logger failed")
    return "L4:log_error"
